### ISSUE TITLE
Relay Pulses on Power-up

### ISSUE THREAD

**Issue by sephorous** [DanielOgorchock/ST_Anything#120]
I'm not really sure if this counts as an "issue" but here it goes. As suggested in a previous issue post, I'm using a modified Sonoff device to control a garage door and get back status. I am using the DoorControl function to accomplish this. Its working *almost* perfectly.  

My only is that when the power cycles on the device, it pulses the relay, causing the garage door to open up. I'm not sure if this is something I can change in the code to avoid this because a power fluctuation will essentially cause my garage door to open up. 

Im mostly using the code from a DoorControl example, not changing any of the default inputs other than whats required. I need to know which part of the code to focus in to keep the relay from pulsing on power up.

Thoughts?

**Comment by ogiewon** [comment]
Which pins are you using on the ESP8266 to control the garage door relays?  I have found, during my testing, that certain pins are toggled during the boot up process of an ESP8266, BEFORE THE ARDUINO SKETCH IS EVEN RUN.  IF you look at my ReadMe, there is a table of pins and their behavior/limitations during a boot process.  If you want to share you sketch, I'd be happy to look it over to see if anything obvious stands out.

I have used both Arduinos and ESP8266s and do not see the relay outputs toggle on boot up if the ST_Anything sketch is configured correctly, and if I avoid the troublesome pins on the ESP8266.

I do not have an Sonoff devices to test with, unfortunately.  You may be running into a limitation of the device?

**Comment by ogiewon** [comment]
Closing due to inactivity.

**Comment by hjung110** [comment]
I am new with this library but I had to use the inverted logic only with a wemos d1 mini paired up with the wemos relay shield. Seems like this does not occur with a separate relay switch. I also couldn't use the garage door sketch with the relay shield because the only pin thats active is the D1 pin to activate the relay so I used the timed relay sketch instead. here is the documentation from the timed relay library for reference 
//			  st::S_TimedRelay() constructor requires the following arguments
//				- String &name - REQUIRED - the name of the object - must match the Groovy ST_Anything DeviceType tile name
//				- byte pinOutput - REQUIRED - the Arduino Pin to be used as a digital output
//				- bool startingState - REQUIRED - the value desired for the initial state of the switch.  LOW = "off", HIGH = "on"
//				- bool invertLogic - REQUIRED - determines whether the Arduino Digital Ouput should use inverted logic
//				- long onTime - REQUIRED - the number of milliseconds to keep the output on, DEFGAULTS to 1000 milliseconds
//				- long offTime - OPTIONAL - the number of milliseconds to keep the output off, DEFAULTS to 0
//				- intnumCycles - OPTIONAL - the number of times to repeat the on/off cycle, DEFAULTS to 1
